### PR TITLE
_checkRequestMethods should throw MethodNotAllowedException

### LINKED ISSUE #409: _checkRequestMethods should throw MethodNotAllowedException
CakePHP has an exception MethodNotAllowedException that returns the HTTP error code 405 (Method Not Allowed). It seems to me that **ApiListener::_checkRequestMethods** should use this exception and error code rather than the more generic BadRequestException (with its standard error code of 400).

### PR DESCRIPTION
**Pull request by Spriz**
Closes #409

### CODE REVIEW

**Comment by ADmad** [comment]
Tests failing.

File: src/Listener/ApiListener.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ protected function _checkRequestMethods()
             }
         }
 
-        throw new BadRequestException('Wrong request method');
+        throw new MethodNotAllowedException('Wrong request method');
```
Comment (ADmad) [inline src/Listener/ApiListener.php]: The message string can also be removed and instead just let the default message for this exception be used.

**Comment by Spriz** [comment]
Oh the joy of editing code from Github 👍

**Comment by Spriz** [comment]
there we go @admad :)

**Pull request merged**